### PR TITLE
Update to Spring-Boot 1.5.3

### PR DESCRIPTION
**Pull request by hendrikebbers**
Changes Spring-Boot dependency to 1.5.3.RELEASE

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/492)

### CODE REVIEW

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformApplication.java [hunk 1]
```diff
@@ -15,20 +15,39 @@
...
-@SpringBootApplication
+@Target({TYPE})
+@Retention(RUNTIME)
+@Documented
+@Inherited
+@SpringBootConfiguration
+@EnableAutoConfiguration
+@ComponentScan(excludeFilters = {
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = TypeExcludeFilter.class),
+        @ComponentScan.Filter(type = FilterType.CUSTOM, classes = AutoConfigurationExcludeFilter.class) })
```
Comment (timo0) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformApplication.java]: Is this a new thing? If you use @AutoConfigurationExcludeFilter at your component that will be not affected by the component scan?

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformSpringBootstrap.java [hunk 2]
```diff
@@ -68,4 +72,14 @@ private void updateConfigurationBySpring(DolphinPlatformConfiguration configurat
...
         }
     }
+
+    private static ApplicationContext ctx = null;
+
+    public static ApplicationContext getApplicationContext() {
+        return ctx;
+    }
+
+    public void setApplicationContext(ApplicationContext ctx) throws BeansException {
```
Comment (timo0) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformSpringBootstrap.java]: where will be the setter called?

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformApplication.java (hunk 1 above)
Comment (hendrikebbers) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformApplication.java]: This is exactly the same as `@SpringBootApplication` is defined in the new Spring Boot version. Based on this I copied it ;)

File: platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformSpringBootstrap.java (hunk 2 above)
Comment (hendrikebbers) [inline platform/dolphin-platform-server-spring/src/main/java/com/canoo/dolphin/server/spring/DolphinPlatformSpringBootstrap.java]: This method is defined in the `ApplicationContextAware` interface and will automatically be called by Spring.

**Pull request merged**